pkg/util/collectionutil.go

```
@@ -0,0 +1,23 @@
+package util
+
+import "fmt"
+
+func AppendMap[KEY comparable, VALUE any](first map[KEY]VALUE, second map[KEY]VALUE) map[KEY]VALUE {
+	for key, value := range second {
+		if _, ok := first[key]; ok {
+			panic(fmt.Sprintf("%v of second (value %v) would overwrite the value of first (%v)", key, value, first[key]))
```

is there a reason we panic here instead of returning an error and shutting down "gracefully"?

---

It is a programmers error. The other are configuration (or user) error.
I changed it.